sbin/functions.sh

```diff
@@ -966,6 +966,189 @@ clean_up()
 	[ -n "${mnt2}" ] && umount ${mnt2}
 }
 
+interactive_config()
+{
+	local device="$1"
+	local localconf_dir="$2"
+	local tmpfile=$(mktemp)
+
+	if [ -z ${device} ] || [ -z ${localconf_dir} ]
+	then
+		debugmsg ${DEBUG_CRIT} "ERROR: interactive_config input parameters not provided"
+		return 1
+	fi
+
+	local localconf="${localconf_dir}/config.sh"
+
+	echo "" > ${localconf}
+
+	# Distribution
+	dialog --no-cancel --inputbox "Distribution name:" 8 40 "OverC"
```

Comment:
I won't go through all the dialogs themselves, but the approach is good. The only thing that I'd suggest is that the questions/defaults really would b e better not coded into the script itself.

Fragments that we could load / read and then execute will make updating and extending it easier.